codes.go

```diff
@@ -204,1478 +756,2101 @@ const (
+	KEY_PROG3:            "KEY_PROG3",
+	KEY_PROG4:            "KEY_PROG4",
+	KEY_ALL_APPLICATIONS: "KEY_ALL_APPLICATIONS",
+	// KEY_DASHBOARD: "KEY_DASHBOARD", // (KEY_ALL_APPLICATIONS)
+	KEY_SUSPEND:        "KEY_SUSPEND",
+	KEY_CLOSE:          "KEY_CLOSE",
+	KEY_PLAY:           "KEY_PLAY",
+	KEY_FASTFORWARD:    "KEY_FASTFORWARD",
+	KEY_BASSBOOST:      "KEY_BASSBOOST",
+	KEY_PRINT:          "KEY_PRINT",
+	KEY_HP:             "KEY_HP",
+	KEY_CAMERA:         "KEY_CAMERA",
+	KEY_SOUND:          "KEY_SOUND",
+	KEY_QUESTION:       "KEY_QUESTION",
+	KEY_EMAIL:          "KEY_EMAIL",
+	KEY_CHAT:           "KEY_CHAT",
+	KEY_SEARCH:         "KEY_SEARCH",
+	KEY_CONNECT:        "KEY_CONNECT",
+	KEY_FINANCE:        "KEY_FINANCE",
+	KEY_SPORT:          "KEY_SPORT",
+	KEY_SHOP:           "KEY_SHOP",
+	KEY_ALTERASE:       "KEY_ALTERASE",
+	KEY_CANCEL:         "KEY_CANCEL",
+	KEY_BRIGHTNESSDOWN: "KEY_BRIGHTNESSDOWN",
+	KEY_BRIGHTNESSUP:   "KEY_BRIGHTNESSUP",
+	KEY_MEDIA:          "KEY_MEDIA",
+
+	KEY_SWITCHVIDEOMODE: "KEY_SWITCHVIDEOMODE",
+	KEY_KBDILLUMTOGGLE:  "KEY_KBDILLUMTOGGLE",
+	KEY_KBDILLUMDOWN:    "KEY_KBDILLUMDOWN",
+	KEY_KBDILLUMUP:      "KEY_KBDILLUMUP",
+
+	KEY_SEND:        "KEY_SEND",
+	KEY_REPLY:       "KEY_REPLY",
+	KEY_FORWARDMAIL: "KEY_FORWARDMAIL",
+	KEY_SAVE:        "KEY_SAVE",
+	KEY_DOCUMENTS:   "KEY_DOCUMENTS",
+
+	KEY_BATTERY: "KEY_BATTERY",
+
+	KEY_BLUETOOTH: "KEY_BLUETOOTH",
+	KEY_WLAN:      "KEY_WLAN",
+	KEY_UWB:       "KEY_UWB",
+
+	KEY_UNKNOWN: "KEY_UNKNOWN",
+
+	KEY_VIDEO_NEXT:       "KEY_VIDEO_NEXT",
+	KEY_VIDEO_PREV:       "KEY_VIDEO_PREV",
+	KEY_BRIGHTNESS_CYCLE: "KEY_BRIGHTNESS_CYCLE",
+	KEY_BRIGHTNESS_AUTO:  "KEY_BRIGHTNESS_AUTO",
+	// KEY_BRIGHTNESS_ZERO: "KEY_BRIGHTNESS_ZERO", // (KEY_BRIGHTNESS_AUTO)
+	KEY_DISPLAY_OFF: "KEY_DISPLAY_OFF",
 
-var REPName = map[EvCode]string{
-	0: "REP_DELAY",
-	1: "REP_PERIOD",
+	KEY_WWAN: "KEY_WWAN",
+	// KEY_WIMAX: "KEY_WIMAX", // (KEY_WWAN)
+	KEY_RFKILL: "KEY_RFKILL",
+
+	KEY_MICMUTE: "KEY_MICMUTE",
+
+	BTN_MISC: "BTN_MISC",
+	// BTN_0: "BTN_0", // (BTN_MISC)
+	BTN_1: "BTN_1",
+	BTN_2: "BTN_2",
+	BTN_3: "BTN_3",
+	BTN_4: "BTN_4",
+	BTN_5: "BTN_5",
+	BTN_6: "BTN_6",
+	BTN_7: "BTN_7",
+	BTN_8: "BTN_8",
+	BTN_9: "BTN_9",
+
+	BTN_MOUSE: "BTN_MOUSE",
+	// BTN_LEFT: "BTN_LEFT", // (BTN_MOUSE)
+	BTN_RIGHT:   "BTN_RIGHT",
+	BTN_MIDDLE:  "BTN_MIDDLE",
+	BTN_SIDE:    "BTN_SIDE",
+	BTN_EXTRA:   "BTN_EXTRA",
+	BTN_FORWARD: "BTN_FORWARD",
+	BTN_BACK:    "BTN_BACK",
+	BTN_TASK:    "BTN_TASK",
+
+	BTN_JOYSTICK: "BTN_JOYSTICK",
+	// BTN_TRIGGER: "BTN_TRIGGER", // (BTN_JOYSTICK)
+	BTN_THUMB:  "BTN_THUMB",
+	BTN_THUMB2: "BTN_THUMB2",
+	BTN_TOP:    "BTN_TOP",
+	BTN_TOP2:   "BTN_TOP2",
+	BTN_PINKIE: "BTN_PINKIE",
+	BTN_BASE:   "BTN_BASE",
+	BTN_BASE2:  "BTN_BASE2",
+	BTN_BASE3:  "BTN_BASE3",
+	BTN_BASE4:  "BTN_BASE4",
+	BTN_BASE5:  "BTN_BASE5",
+	BTN_BASE6:  "BTN_BASE6",
+	BTN_DEAD:   "BTN_DEAD",
+
+	BTN_GAMEPAD: "BTN_GAMEPAD",
+	// BTN_SOUTH: "BTN_SOUTH", // (BTN_GAMEPAD)
+	// BTN_A: "BTN_A", // (BTN_GAMEPAD)
```

Comment:
`XToString` currently points to one value only if someone wants to convert it back to value with `XFromString` for some reason.

Maybe it would be a good idea to make use of `XName` as informative string only with merged duplicates like
```go
var KEYToString = map[EvCode]string{
    BTN_GAMEPAD: "BTN_GAMEPAD/BTN_SOUTH/BTN_A",
}
```

Reply:
Done.

Let me know if you would like to remove it.  
Maybe it would be also a good idea to split these groups/mappings to separate files.